src/themes/default/components/core/AddToCart.vue

```diff
@@ -18,8 +18,8 @@ export default {
     onAfterRemovedVariant () {
       this.$forceUpdate()
     },
-    canBeAdded (product) {
-      return formatProductMessages(product.errors) !== ''
+    cantBeAdded (product) {
+      return formatProductMessages(product.errors) !== '' || this.disabled
```

Comment:
This function should be named `isProductDisabled` rather than `canBeAdded` which is miss-leading. I know that it wasn't a part of Your PR but please do fix it by this chance :)